src/briefcase/integrations/android_sdk.py

```diff
@@ -212,61 +248,305 @@ def verify_emulator(self):
+                running_avds[avd] = d
+                full_name = "@{avd} (running {name} emulator)".format(
+                    avd=avd, name=name,
+                )
+                choices.append((d, full_name))
+
+                # Save the AVD as a device detail.
+                details["avd"] = avd
+
+                # Device can be looked up by device ID or AVD
+                device_choices[d] = full_name
+                device_choices["@" + avd] = full_name
+            else:
+                # It's a physical device (might be disabled)
+                choices.append((d, name))
+                device_choices[d] = name
+
+        # Add any non-running emulator AVDs to the list of candidate devices
+        for avd in self.emulators():
+            if avd not in running_avds:
+                name = "@{avd} (emulator)".format(avd=avd)
+                choices.append(("@" + avd, name))
+                device_choices["@" + avd] = name
+
+        # If a device or AVD has been provided, check it against the available
+        # device list.
+        if device_or_avd:
+            try:
+                name = device_choices[device_or_avd]
+
+                if device_or_avd.startswith("@"):
+                    # specifier is an AVD
+                    try:
+                        avd = device_or_avd[1:]
+                        device = running_avds[avd]
+                    except KeyError:
+                        # device_or_avd isn't in the list of running avds;
+                        # it must be a non-running emulator.
+                        return None, name, avd
+                else:
+                    # Specifier is a direct device ID
+                    avd = None
+                    device = device_or_avd
+
+                details = running_devices[device]
+                avd = details.get("avd")
+                if details["authorized"]:
+                    # An authorized, running device (emulator or physical)
+                    return device, name, avd
+                else:
+                    # An unauthorized physical device
+                    raise AndroidDeviceNotAuthorized(device)
+
+            except KeyError:
+                # Provided device_or_id isn't a valid device identifier.
+                if device_or_avd.startswith("@"):
+                    id_type = "emulator AVD"
+                else:
+                    id_type = "device ID"
+                raise InvalidDeviceError(id_type, device_or_avd)
+
+        # We weren't given a device/AVD; we have to select from the list.
+        # If we're selecting from a list, there's always one last choice
+        choices.append((None, "Create a new Android emulator"))
+
+        # Show the choices to the user.
+        print()
+        print("Select device:")
+        print()
+        choice = select_option(choices, input=self.command.input)
+
+        # Proces the user's choice
+        if choice is None:
```

Comment:
Dumb question: is this the only way to get `None` out of `select_option()`? Does it have a way to e.g. cancel, which would return `None`? I just fear using `None` as a sentinel value if it might be actually returned.

Reply:
A fair point; but select_choice always returns one of the choices, or raises an exception.